firestore/src/common/document_change.cc

```diff
@@ -23,6 +23,8 @@ using Type = DocumentChange::Type;
 // Older NDK (r16b) fails to define this properly. Fix this when support for
 // the older NDK is removed.
 const std::size_t DocumentChange::npos = static_cast<std::size_t>(-1);
+#else
+constexpr std::size_t DocumentChange::npos;
```

Comment:
@var-const Do I have to initialize `npos` here, as well as in the header file? I did some research about `constexpr` and couldn't find a definitive answer. The tests that I added work fine, but it's possible that they're just inlining the value of `npos` from the header file.

Reply:
Nevermind, it's a compiler error to re-define it here. I thought I had tried that. Obviously not lol.

Reply:
If you're interested, this is something that changed in C++17 -- before C++17, defining a static `constexpr` variable like this was necessary, but starting from C++17 it's unnecessary and deprecated (see the last code snippet and the text around it [here](https://en.cppreference.com/w/cpp/language/static)). Like you said, this issue didn't come up previously because the variable wasn't used ("odr-used"). This [TotW](go/totw/140#a-static-class-member) is also relevant.